sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala

```diff
@@ -46,4 +46,10 @@ class ExperimentalMethods private[sql]() {
 
   @volatile var extraOptimizations: Seq[Rule[LogicalPlan]] = Nil
 
+  override def clone(): ExperimentalMethods = {
```

Comment:
When I reviewing the code, I just found they are not thread-safe, although they have already been declared volatile. 

I am fine to keep them unchanged. How about leaving a comment in the code to emphasize that it is not thread-safe.